app/src/test/java/backgammon/DoublingCubeTests.java

```diff
@@ -20,6 +23,43 @@ public static void main(String[] args){
     jim.setCubeValue(64);
 
     jim.cubeCheck(guy.getCubeOwnership());
+    */
+
+    TestPlayerDoublingCube guy1 = new TestPlayerDoublingCube(true);
+    TestPlayerDoublingCube guy2 = new TestPlayerDoublingCube(true);
+
+    //Should be yes, no, yes, no
+    if(jim.cubeCheck(guy1.getCubeOwnership())){
+      jim.rollCube();
+      guy1.setCubeOwnership(false);
+    }
+    else
+      System.out.println("Didn't thing");
+
+    if(jim.cubeCheck(guy1.getCubeOwnership())){
+      jim.rollCube();
+      guy1.setCubeOwnership(false);
+    }
+    else
+      System.out.println("Didn't thing");
+
+    guy2.setCubeOwnership(jim.rollCheck());
+
+    if(jim.cubeCheck(guy2.getCubeOwnership())){
+      jim.rollCube();
+      guy1.setCubeOwnership(false);
+    }
+    else
+      System.out.println("Didn't thing");
+
+    guy2.setCubeOwnership(jim.rollCheck());
+    guy1.setCubeOwnership(jim.rollCheck());
 
+    if(jim.cubeCheck(guy1.getCubeOwnership())){
+      jim.rollCube();
+      guy1.setCubeOwnership(false);
+    }
+    else
+      System.out.println("Didn't thing");
```

Comment:
You don't understand how to test it properly.
Don't use if statements, use assertEquals.
If you set up your IDE properly you should even be getting tab completion for these methods.
If you didn't set it up right, read the instructions [here](https://github.com/jmdaemon/backgammon#readme) and do it properly.

Reply:
Either you or me will have to rewrite this entire thing.